src/vs/workbench/electron-browser/main.ts

```diff
@@ -75,6 +77,33 @@ export function startup(configuration: IWindowConfiguration): TPromise<void> {
 	return openWorkbench(configuration);
 }
 
+function revive(workbench: IWindowConfiguration) {
+	if (workbench.folderUri) {
+		workbench.folderUri = uri.revive(workbench.folderUri);
+	}
+	function reviveFileUri(path: { fileUri?: UriComponents }) {
+		if (path.fileUri) {
+			path.fileUri = uri.revive(path.fileUri);
+		}
+	}
+	const filesToOpen = workbench.filesToOpen;
```

Comment:
@aeschli that is a lot of ugly repeated code, can you not just throw in a loop and reduce this method to 5 lines? 

`[workbench.filesToOpen, workbench.filesToCreate, workbench.filesToDiff, workbench.filesToWait].forEach(...)`

Reply:
👍